Sources/StreamChat/WebSocketClient/EventMiddlewares/MemberEventMiddleware.swift

```diff
@@ -7,33 +7,56 @@ import Foundation
 /// The middleware listens for `MemberEvent`s and updates `ChannelDTO`s accordingly.
 struct MemberEventMiddleware<ExtraData: ExtraDataTypes>: EventMiddleware {
     func handle(event: Event, session: DatabaseSession) -> Event? {
-        guard let memberEvent = event as? MemberEvent else { return event }
-        
         do {
-            switch memberEvent {
+            switch event {
             case is MemberAddedEvent, is MemberUpdatedEvent:
                 guard let eventWithMemberPayload = event as? EventWithMemberPayload,
                       let eventPayload = eventWithMemberPayload.payload as? EventPayload<ExtraData>,
                       let memberPayload = eventPayload.memberContainer?.member
                 else {
                     break
                 }
-                try session.saveMember(payload: memberPayload, channelId: memberEvent.cid)
-            case is MemberRemovedEvent:
-                guard let channel = session.channel(cid: memberEvent.cid) else {
+                try session.saveMember(payload: memberPayload, channelId: (event as! EventWithChannelId).cid)
```

Comment:
`event as! EventWithChannelId` are we sure this is always true? Why the force unwrap? 🤔

Reply:
`MemberRemovedEvent` is always `EventWithChannelId` :) @nuno-vieira